nexus-sdk-js/src/View/ElasticSearchView/index.ts

```diff
@@ -0,0 +1,175 @@
+  };
+  timed_out: boolean;
+  took: number;
+}
+
+class ElasticSeachView {
+  id: string;
+  type: string[];
+  uuid: string;
+  includeMetadata: boolean;
+  mapping: Mapping;
+  sourceAsText: boolean;
+  rev: number;
+  deprecated: boolean;
+  orgLabel: string;
+  projectLabel: string;
+  queryURL: string;
+  constructor(
+    orgLabel: string,
+    projectLabel: string,
+    elasticSearchViewResponse: ElasticSearchViewResponse,
+  ) {
+    this.orgLabel = orgLabel;
+    this.projectLabel = projectLabel;
+    this.id = elasticSearchViewResponse['@id'];
+    this.type = elasticSearchViewResponse['@type'];
+    this.uuid = elasticSearchViewResponse['_uuid'];
+    this.includeMetadata = elasticSearchViewResponse['includeMetadata'];
+    this.mapping = elasticSearchViewResponse['mapping'];
+    this.sourceAsText = elasticSearchViewResponse['sourceAsText'];
+    this.rev = elasticSearchViewResponse['_rev'];
+    this.deprecated = elasticSearchViewResponse['_deprecated'];
+    this.queryURL = `views/${this.orgLabel}/${this.projectLabel}/${
+      this.id
+    }/_search`;
+  }
+
+  /**
+   * Convenience method to help in filtering a project by @Types
+   *
+   * @param {string[]} types
+   * @param {PaginationSettings} [pagination]
+   * @returns {Promise<PaginatedList<Resource>>}
+   * @memberof ElasticSeachView
+   */
+  async filterByTypes(
+    types: string[],
+    pagination?: PaginationSettings,
+  ): Promise<PaginatedList<Resource>> {
+    const query = {
+      query: {
+        bool: {
+          filter: types.map(type => ({ term: { '@type': type } })),
+        },
+      },
+    };
+    return await this.query(query, pagination);
+  }
+
+  /**
+   * Convenience method to help in filtering a project by what
+   * Schema that the resources are conforming to
+   *
+   * @param {string} constrainedBy
+   * @param {PaginationSettings} [pagination]
+   * @returns {Promise<PaginatedList<Resource>>}
+   * @memberof ElasticSeachView
+   */
+  async filterByConstrainedBy(
+    constrainedBy: string,
+    pagination?: PaginationSettings,
+  ): Promise<PaginatedList<Resource>> {
+    const query = {
+      query: {
+        term: { _constrainedBy: constrainedBy },
+      },
+    };
+    return await this.query(query, pagination);
+  }
+
+  /**
+   * Search for resources located in a project
+   * using an elastic search Query (as a JS Object) as input
+   * https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html
+   *
+   * @param {Object} elasticSearchQuery
+   * @param {PaginationSettings} [pagination]
+   * @returns {Promise<PaginatedList<Resource>>}
+   * @memberof ElasticSeachView
+   */
+  async query(
+    elasticSearchQuery: Object,
+    pagination?: PaginationSettings,
+  ): Promise<PaginatedList<Resource>> {
+    const requestURL = pagination
+      ? `${this.queryURL}?from=${pagination.from}&size=${pagination.size}`
+      : this.queryURL;
+    const response: ElasticSearchViewQueryResponse = await httpPost(
```

Comment:
Remember than `httpPost` can crash, so you should wrap this in a `try/catch` statement. We need to change the signature of our `http` method and make them return `Promise<T | Error>` so typescript can catch stuff like that.